k8s/operators/src/diskpool/main.rs

```diff
@@ -619,13 +632,34 @@ impl ResourceContext {
 /// is wrong would be to consult the logs.
 async fn ensure_crd(k8s: Client) {
     let dsp: Api<k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition> = Api::all(k8s);
-    let lp = ListParams::default().fields(&format!("metadata.name={}", "diskpools.openebs.io"));
-    let crds = dsp.list(&lp).await.expect("failed to list CRDS");
 
-    // the CRD has not been installed yet, to avoid overwriting (and create upgrade issues) only
-    // install it when there is no crd with the given name
-    if crds.iter().count() == 0 {
-        let crd = DiskPool::crd();
+    // Check if there is an existing dsp crd. If yes, Replace it with new generated one.
+    // Create new if it doesnt exist.
+    let mut crd = DiskPool::crd();
+    let crd_name = crd.metadata.name.as_ref().unwrap().as_str();
+    if let Ok(existing) = dsp.get(crd_name).await {
+        crd.metadata.resource_version = existing.resource_version();
+        info!(
+            "Replacing CRD: {}",
+            serde_json::to_string_pretty(&crd).unwrap()
+        );
+
+        let pp = PostParams::default();
+        match dsp.replace(crd_name, &pp, &crd).await {
+            Ok(o) => {
+                info!(crd = ?o.name_any(), "created");
+                // let the CRD settle this purely to avoid errors messages in the console
+                // that are harmless but can cause some confusion maybe.
+                tokio::time::sleep(Duration::from_secs(5)).await;
+            }
+
+            Err(e) => {
+                error!("failed to create CRD error {}", e);
+                tokio::time::sleep(Duration::from_secs(1)).await;
+                std::process::exit(1);
+            }
```

Comment:
seems this code is duplicated below, can we reuse?

Reply:
Incorporated